golos.publication/golos.publication.cpp

```diff
@@ -129,9 +131,27 @@ void publication::create_message(name account, std::string permlink,
         eosio_assert(prop_sum <= config::_100percent, "publication::create_message: prop_sum > 100%");
         benefic_map[ben.account] += ben.deductprcnt; //several entries for one user? ok.
     }
-    eosio_assert((benefic_map.size() <= max_beneficiaries_param.max_beneficiaries), "publication::create_message: benafic_map.size() > MAX_BENEFICIARIES");
 
-    //reusing a vector
+    if (referral_acc_param.account != name()) {
+        eosio_assert(is_account(referral_acc_param.account), "Referral account doesn't exist.");
```

Comment:
You can remove this check. It's so expensive make request to database every post. Moreover this account was checked on set parameters.